cms/djangoapps/contentstore/views/course.py

```diff
@@ -451,11 +475,15 @@ def filter_ccx(course_access):
                 # If a user has access to a course that doesn't exist, don't do anything with that course
                 pass
 
+            if split_archived and course.has_ended():
```

Comment:
Judging from line 482, `course` can be `ErrorDescriptor` in some case, which is unlikely to have a `has_ended` method, so this would error out.  I think we need the following logic:
```
if course is not None and not isinstance(course, ErrorDescriptor):
    if split_archived and course.has_ended():
        archived_courses[course_key] = course
    else:
        active_courses[course_key] = course
```